### PR TITLE
Convert homogentable to array/vector/matrix

### PR DESCRIPTION
**Pull request by minmingzhu**
## What changes were proposed in this pull request?

1. Convert homogentable to array/vector/matrix
2. add unit test

## Does this PR also require the following changes?

- CI
- Documentation
- Example

No.

### CODE REVIEW

File: mllib-dal/src/main/native/oneapi/dal/ColumnAccessorImpl.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ * Method:    cPull
+ * Signature: (JJ)[D
+ */
+JNIEXPORT jdoubleArray JNICALL Java_com_intel_oneapi_dal_table_ColumnAccessor_cPullDouble
+  (JNIEnv *env, jobject, jlong cTableAddr, jlong cColumnIndex, jlong cRowStartIndex, jlong cRowEndIndex) {
+      homogen_table *htable =
+              ((std::shared_ptr<homogen_table> *)cTableAddr)->get();
+      column_accessor<const double> *acc = new column_accessor<const double>(*htable);
+      const auto col_values = acc->pull(cColumnIndex, {cRowStartIndex, cRowEndIndex});
+      double* col_doubles = new double[col_values.get_count()];
```
Comment (xwu99) [inline mllib-dal/src/main/native/oneapi/dal/ColumnAccessorImpl.cpp]: you don't need to create extra buffer and duplicate col_values. it's a waste for memory transfer.
Could you find if the JNI function you use has some "isCopy" flag to copy the buffer?
Also just dont use new without delete in C++, it's recommanded to use smart pointer.

File: mllib-dal/src/main/native/oneapi/dal/HomogenTableImpl.cpp [hunk 2]
```diff
@@ -134,10 +134,10 @@ JNIEXPORT jlong JNICALL Java_com_intel_oneapi_dal_table_HomogenTableImpl_lInit(
  */
 JNIEXPORT jlong JNICALL
 Java_com_intel_oneapi_dal_table_HomogenTableImpl_cGetColumnCount(
-    JNIEnv *env, jobject, jlong ctableAddr) {
+    JNIEnv *env, jobject, jlong cTableAddr) {
     printf("HomogenTable getcolumncount \n");
```
Comment (xwu99) [inline mllib-dal/src/main/native/oneapi/dal/HomogenTableImpl.cpp:276]: could you remove all printf debug if it's unnecessay.

**Review by xwu99 — CHANGES_REQUESTED** [review]

**Comment by xwu99** [comment]
@minmingzhu could you rebase to master then I can review this PR.

File: mllib-dal/src/main/java/com/intel/oneapi/dal/table/ColumnAccessor.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    public double[] pullDouble(long columnIndex, Common.ComputeDevice device){
+        return this.cPullDouble(this.cObject, columnIndex, 0, -1, device.ordinal());
+    }
+
+    public double[] pullDouble(long columnIndex, long rowStartIndex, long rowEndIndex, Common.ComputeDevice device){
+        return this.cPullDouble(this.cObject, columnIndex, rowStartIndex, rowEndIndex, device.ordinal());
+    }
+
+    private native double[] cPullDouble(long cObject, long cColumnIndex,
```
Comment (xwu99) [inline mllib-dal/src/main/java/com/intel/oneapi/dal/table/ColumnAccessor.java:18]: Pls put all public interfaces before all private members.

File: mllib-dal/src/main/native/GPU.cpp [hunk 4]
```diff
@@ -6,11 +6,25 @@
...
 static std::vector<std::shared_ptr<sycl::queue>> cVector;
-sycl::queue *getQue() {
+
+static void setVector(std::shared_ptr<sycl::queue> *ptr) {
+    mtx.lock();
+    cVector.push_back(*ptr);
+    mtx.unlock();
+}
+
+static sycl::queue *getQue(const sycl::device device) {
```
Comment (xwu99) [inline mllib-dal/src/main/native/GPU.cpp:16]:
```suggestion
static sycl::queue *getSyclQueue(const sycl::device device) {
```

Comment (xwu99) [inline mllib-dal/src/main/native/GPU.cpp:16]: pls use refence (sycl::queue &) instead of pointer.

File: mllib-dal/src/main/native/GPU.cpp [hunk 5]
```diff
@@ -6,11 +6,25 @@
...
+    mtx.lock();
     if (!cVector.empty()) {
         return cVector[0].get();
+    } else {
+        sycl::queue *queue = new sycl::queue(device);
+        std::shared_ptr<sycl::queue> *queuePtr =
+            new std::shared_ptr<sycl::queue>(queue);
```
Comment (xwu99) [inline mllib-dal/src/main/native/GPU.cpp:23]: you need to use std::make_shared of C++ library.

File: mllib-dal/src/main/native/oneapi/dal/HomogenTableImpl.cpp (hunk 2 above)
Comment (minmingzhu) [reply mllib-dal/src/main/native/oneapi/dal/HomogenTableImpl.cpp]: I will delete the all printf debug code at the last branch.

**Comment by xwu99** [comment]
@minmingzhu  Pls merge the PR. 
and select "Squash and Merge" instead of "Merge pull request"

**Comment by minmingzhu** [comment]
merge convert_homogentable to dpcpp

**Pull request merged**